### PR TITLE
✨ circulating supply endpoint

### PR DESCRIPTION
**Pull request by sebipap**
got `132277978350310331891917` as result (~132K)

### CODE REVIEW

File: pages/api/circulating-supply.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  timelockController,
+  rewardsController,
+  escrowedEXA,
+  TREASURY,
+  airdrop,
+];
+
+const client = createPublicClient({
+  chain: optimism,
+  transport: http(),
```
Comment (cruzdanilo) [inline pages/api/circulating-supply.ts:27]: use our alchemy rpc instead of public one

**Review by franm91 — APPROVED** [review]

File: pages/api/circulating-supply.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+});
+
+const exa = {
+  abi: exaABI,
+  address: exaAddress as `0x${string}`,
```
Comment (cruzdanilo) [inline pages/api/circulating-supply.ts:32]: `getAddress(exaAddress)`

**Review by cruzdanilo — CHANGES_REQUESTED** [review]

File: pages/api/circulating-supply.ts (hunk 2 above)
Comment (sebipap) [reply pages/api/circulating-supply.ts:32]: cool

**Review by itofarina — APPROVED** [review]

**Review by jgalat — APPROVED** [review]
lgtm,
1 minor detail is that this endpoint will work on all envs (i.e. `ethereum.exact.ly/api/circulating-supply`). I don't have a problem with that but just highlighting it in case anyone thinks otherwise

File: pages/api/circulating-supply.ts [hunk 3]
```diff
@@ -31,6 +31,8 @@ const exa = {
 } as const;
 
 export default async function (req: NextApiRequest, res: NextApiResponse) {
+  res.setHeader('Cache-Control', 's-maxage=86400');
+  console.log('CACHE MISS');
```
Comment (cruzdanilo) [inline pages/api/circulating-supply.ts:35]: this only enables cache in the client. we need caching on the server side.

**Review by cruzdanilo — CHANGES_REQUESTED** [review]

File: pages/api/circulating-supply.ts (hunk 3 above)
Comment (sebipap) [reply pages/api/circulating-supply.ts]: as discussed, this is true, but in this case the "client" is an API gateway that handles caching server side.

**Review by jgalat — APPROVED** [review]

**Review by cruzdanilo — APPROVED** [review]

**Review by itofarina — APPROVED** [review]

**Pull request merged**